core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -688,7 +696,13 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
       loadMetadataIfNotExistAndJournal(inodePath,
           LoadMetadataOptions.defaults().setCreateAncestors(true), journalContext);
       ensureFullPathAndUpdateCache(inodePath);
+      srcInode = inodePath.getInode();
       return getFileInfoInternal(inodePath);
+    } catch (AccessControlException e) {
+      rpcContext.setAllowed(false);
+      throw e;
+    } finally {
+      checkAndCommitAuditLog("getFileInfo", path.toString(), null, srcInode, rpcContext);
```

Comment:
Could we make this only take `RpcContext` and update the fields in `RpcContext`?